From da9f568ea76766be3757068540be2b7c8d4405f9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Wed, 24 Jan 2018 17:18:35 +0000
Subject: [PATCH] gtk3: single click with selection set doesn't unselect

this is a safe backportable fix, the unconditional

nFlags |= SelectionEngineFlags::CMDEVT

looks the real dubious thing to me, but only the gtk3 backend has swipe support
so just returning immediately on a swipe definitely ok

Change-Id: Iade26d0d9260ce76dfb2766ac9e5ca9fa9275bf4
---
 vcl/source/window/seleng.cxx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vcl/source/window/seleng.cxx b/vcl/source/window/seleng.cxx
index c911a9059551..c545824436cd 100644
--- a/vcl/source/window/seleng.cxx
+++ b/vcl/source/window/seleng.cxx
@@ -342,7 +342,7 @@ void SelectionEngine::Reset()
 void SelectionEngine::Command( const CommandEvent& rCEvt )
 {
     // Timer aWTimer is active during enlarging a selection
-    if ( !pFunctionSet || !pWin || aWTimer.IsActive() )
+    if ( !pFunctionSet || !pWin || aWTimer.IsActive() || rCEvt.GetCommand() == CommandEventId::Swipe )
         return;
     aWTimer.Stop();
     nFlags |= SelectionEngineFlags::CMDEVT;
-- 
2.14.3

